src/platyPS/platyPS.psm1

```diff
@@ -599,6 +603,70 @@ function New-MarkdownAboutHelp
     }
 }
 
+function New-YamlHelp
+{
+    [CmdletBinding()]
+    [OutputType([System.IO.FileInfo[]])]
+    param(
+        [Parameter(Mandatory=$true,
+            Position=1,
+            ValueFromPipeline=$true,
+            ValueFromPipelineByPropertyName=$true)]
+        [string[]]$Path,
+
+        [Parameter(Mandatory=$true)]
+        [string]$OutputPath,
+
+        [System.Text.Encoding]$Encoding = [System.Text.Encoding]::UTF8,
+        
+        [switch]$Force
+    )
+	begin 
+	{
+		validateWorkingProvider
+
+		$MarkdownFiles = @()
+
+		if(-not (Test-Path $OutputPath))
+		{
+			mkdir $OutputPath -ErrorAction SilentlyContinue > $null
+		}
+
+		if(-not (Test-Path -PathType Container $OutputPath))
+		{
+			throw "$OutputPath is not a container"
+			return
+		}
+	}
+	process 
+	{
+		$MarkdownFiles += GetMarkdownFilesFromPath $Path
+	}
+	end 
+	{
+		$MarkdownFiles | ForEach-Object {
+            Write-Verbose "[New-ExternalHelp] Input markdown file $_"
+        }
+
+		foreach($markdownFile in $MarkdownFiles)
+		{
+			$mamlModels = GetMamlModelImpl $markdownFile.FullName -Encoding $Encoding
+			if($mamlModels.Count -ne 1)
+			{
+				Write-Error "$($mamlModels.Count) commands registered in $markdownFile insteadn of one which is not supported for Yaml help"
+				continue
+			}
+			$mamlModel = $mamlModels[0]
+			$markdownMetadata = Get-MarkdownMetadata -Path $MarkdownFile.FullName
+			$mamlModel.ModuleName = $markdownMetadata[$script:MODULE_PAGE_MODULE_NAME]
```

Comment:
This is clunky: ModuleName becomes this only mutable part of $mamlModel that we have to set from PS.
It would be cleaner to pass it as a parameter to MamlModelTransformer.